recipes/flann/all/conanfile.py

```diff
@@ -42,7 +42,7 @@ def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("lz4/1.9.4")
+        self.requires("lz4/1.9.4", transitive_headers=True, transitive_libs=True)
```

Comment:
The Conan graph isn't supposed to match cmakes definition of public and doing this is a big warning sign.

The libraries should not be exposed even if the headers are... downstream consumers should explicitly depend on this and not just "hope and a prayer" they are exposed

```suggestion
        self.requires("lz4/1.9.4", transitive_headers=True)
```

Reply:
Please put that in as a comment!!!! This is exactly why the trait exists and we want to encourage the best practice for not expose your dependencies.

I appreciate your expertise but we are always trying to help shape the C++ ecosystem so we are trying to drive the bigger picture

I am all about the "default general advice" and currenty (abosultely it will change) is to not put this in first and added it later

We obviously do not know the source code of all these project and it's okay to fix them later thats totally normal